packages/web-components/src/components/back-to-top/back-to-top.ts

```diff
@@ -0,0 +1,165 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2021
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html, property, customElement, LitElement } from 'lit-element';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings.js';
+import HostListener from 'carbon-web-components/es/globals/decorators/host-listener';
+import HostListenerMixin from 'carbon-web-components/es/globals/mixins/host-listener';
+import settings from 'carbon-components/es/globals/js/settings.js';
+import throttle from 'lodash-es/throttle.js';
+import UpToTop20 from 'carbon-web-components/es/icons/up-to-top/20.js';
+import styles from './back-to-top.scss';
+
+const { prefix } = settings;
+const { stablePrefix: ddsPrefix } = ddsSettings;
+interface Cancelable {
+  cancel(): void;
+}
+
+/**
+ * Back to top
+ *
+ * @element dds-back-to-top
+ */
+@customElement(`${ddsPrefix}-back-to-top`)
+class DDSBackToTop extends HostListenerMixin(LitElement) {
+  /**
+   * The observer for the resize of the viewport.
+   */
+  private _observerResizeRoot: any | null = null; // TODO: Wait for `.d.ts` update to support `ResizeObserver`
+
+  /**
+   * The observer for the resize of the document body.
+   */
+  private _observerResizeBody: any | null = null; // TODO: Wait for `.d.ts` update to support `ResizeObserver`
+
+  /**
+   * The document height
+   */
+  private _bodyHeight!: number;
+
+  /**
+   * The viewport height
+   */
+  private _windowHeight!: number;
+
+  /**
+   * The handler for throttled scrolling
+   */
+  private _throttleScroll: (((event: Event) => void) & Cancelable) | null = null;
+
+  /**
+   * Button click scrolls to top
+   */
+  // eslint-disable-next-line class-methods-use-this
+  private _handleOnClick() {
+    this.ownerDocument!.defaultView!.scrollTo({ top: 0, behavior: 'smooth' });
+  }
+
+  /**
+   * Button visible when one full page length is scrolled if _showBackToTop is true
+   */
+  private _handleOnScroll() {
+    if (this._showBackToTop()) {
+      this.hidden = this.ownerDocument!.documentElement.scrollTop <= this._windowHeight;
+    }
+  }
+
+  /**
+   * Cleans-up and creats the resize observer for the scrolling container.
+   *
+   * @param [options] The options.
+   * @param [options.create] `true` to create the new resize observer.
+   */
+  private _cleanAndCreateObserverResize({ create }: { create?: boolean } = {}) {
+    if (this._observerResizeRoot) {
+      this._observerResizeRoot.disconnect();
+      this._observerResizeRoot = null;
+    }
```

Comment:
Let's add clean-up code for `._observerResizeBody`.

Reply:
Good catch thanks @asudoh 👍